### PR TITLE
[INFRASTRUCTURE] The fault domain count is different per region - application tier fixes

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Deploy the correct number of fault domains for the application tier

## Solution
Persist the fault domain count in a json file and use it for the count

## Tests
Deploy a new environment

## Notes
`    az vm list-skus --resource-type availabilitySets --query '[?name==`Aligned`].{​Location:locationInfo[0].location, MaximumFaultDomainCount:capabilities[0].value}​' -o Table`

avset and VMs are recreated after fault domain count changes: file bug with azurerm [#9577](https://github.com/terraform-providers/terraform-provider-azurerm/issues/9577)

### CODE REVIEW

File: deploy/configs/faults.json [hunk 1]
```diff
@@ -0,0 +1,174 @@
+[
```
Comment (rukawata) [inline deploy/configs/faults.json:1]: Can we name the file so other people can better understand? eg. max_fault_domain_count.json

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf [hunk 2]
```diff
@@ -64,6 +65,12 @@ locals {
   resource_suffixes = var.naming.resource_suffixes
 
   region  = try(var.infrastructure.region, "")
+
+  faultdomain_count = try(tonumber(compact(
+    [for pair in local.faults : 
+      pair.Location == local.region ? pair.MaximumFaultDomainCount : ""
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf:71]: Region may spell with uppercase/lowercase/ or mix...so I would compare with both uppercase/lowercase.

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/infrastructure.tf [hunk 3]
```diff
@@ -126,7 +126,7 @@ resource "azurerm_availability_set" "scs" {
   location                     = var.resource_group[0].location
   resource_group_name          = var.resource_group[0].name
   platform_update_domain_count = 20
-  platform_fault_domain_count  = 2
+  platform_fault_domain_count  = local.faultdomain_count
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/app_tier/infrastructure.tf:129]: SCS should be 2, not the max.

**Review by rukawata — COMMENTED** [review]
Thanks for adding this. See some small comments.

**Comment by rukawata** [comment]
Another thing is I noticed that by changing this number, avset got re-created, hence VMs got re-created.
I'm not sure this is the expected behavior, opening an issue with terraform.

File: deploy/configs/faults.json (hunk 1 above)
Comment (KimForss) [reply deploy/configs/faults.json:1]: done

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/infrastructure.tf (hunk 3 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/app_tier/infrastructure.tf:129]: For instance Woolworths have a standby SCS - making the count 3 - so I would keep the max value here

**Review by rukawata — APPROVED** [review]

**Review by rukawata — APPROVED** [review]

**Pull request merged**